checker/checker.go

```diff
@@ -27,15 +30,15 @@ func New(syntaxTree *node.ProgramNode) *Checker {
 // If errors occur during one of those phases, the process is stopped at the end of the failing phase.
 // Returns the symbol table and errors
 func (c *Checker) Run() (*symbol.SymbolTable, []error) {
-	c.symbolTable, c.errors = phase.RunSymbolConstruction(c.syntaxTree)
+	c.symbolTable, c.errors = symbolconstruction.RunSymbolConstruction(c.syntaxTree)
```

Comment:
You can now simplify the function `symbolconstruction.Run(...)`, since the phases are in different package.

Do it for the rest of the phases as well.